frontend/src/components/posts/buttons/edit-post-button.tsx

```diff
@@ -0,0 +1,35 @@
+"use client"
+
+import { Post } from "@/utils/posts/definitions";
+import EditPostModal from "../modals/edit-post-modal";
+import { useEffect, useState } from "react";
+import styles from "./post-buttons.module.css"
+import { createPortal } from "react-dom";
+import { checkAuthorPrivileges, checkHubPrivileges } from "@/utils/fetchPrivileges";
```

Comment:
Small nitpick but this line can be removed since you aren't using the checkXPrivileges in this file